cls_module/cls_module/memory/stm/aha.py

```diff
@@ -458,17 +478,44 @@ def forward_memory(self, inputs, targets, labels):
 
     outputs['pc'] = self.forward_pc(inputs=pc_cue)
 
-    losses['pm'], outputs['pm'] = self.forward_ae(name='pm', inputs=outputs['pc'], targets=targets)
-    losses['pm_ec'], outputs['pm_ec'] = self.forward_ae(name='pm_ec', inputs=outputs['pc'], targets=targets)
+    if self.config.get('msp_type', None) == 'ca1':
+      # During study, EC will drive the CA1
+      losses['ca1'], outputs['ca1'] = self.forward_ae(name='ca1', inputs=inputs, targets=inputs)
+      ca3_ca1_target = outputs['ca1']['encoding']
+
+      losses['ca3_ca1'], outputs['ca3_ca1'] = self.forward_ae(name='ca3_ca1',
```

Comment:
Do you mean similar to the PR/PS/PC forward methods?

Reply:
Yeah, something like self.
`self.ca1.forward()`

Reply:
Yeah, as I have been adding more sub-modules.. I have been feeling that it is time to actually split them out, rather than keep all the builds/initialisations & forwards all inside the single AHA module

Reply:
We may need to discuss how we want to split/organise them though